apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/CustomElementMatchers.java

```diff
@@ -87,18 +87,19 @@ public boolean matches(@Nullable ClassLoader target) {
     private static boolean canLoadClass(@Nullable ClassLoader target, String className) {
         boolean result;
         try {
-            Class<?> clazz = Class.forName(className, false, target);
+            if (target == null) {
+                target = ClassLoader.getSystemClassLoader();
```

Comment:
```suggestion
                target = ClassLoader.getSystemClassLoader().getParent();
```

This will reduce a lot of false positives, for environments where dependencies we are filtering for were appended to the system classpath. 
You should check that this doesn't yield `null`, but normally it shouldn't.